p2p/node/localnodestore.go

```diff
@@ -23,114 +21,77 @@ type nodeFileData struct {
 
 // PersistData save node's data to local disk in `path`.
 func (n *LocalNode) PersistData(path string) error {
-
 	data := nodeFileData{
 		PubKey:  n.publicKey.String(),
 		PrivKey: n.privKey.String(),
 	}
 
 	finaldata, err := json.MarshalIndent(data, "", "  ")
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to marshal data: %w", err)
 	}
 
 	datadir := filepath.Join(path, config.P2PDirectoryPath, config.NodesDirectoryName, n.publicKey.String())
-
-	err = filesystem.ExistOrCreate(datadir)
-	if err != nil {
-		return err
+	if err = filesystem.ExistOrCreate(datadir); err != nil {
+		return fmt.Errorf("failed to check or create datadir: %w", err)
 	}
 
 	nodefile := filepath.Join(datadir, config.NodeDataFileName)
 
 	// make sure our node file is written to the os filesystem.
-	f, err := os.Create(nodefile)
-	if err != nil {
-		return err
-	}
-
-	_, err = f.Write(finaldata)
-	if err != nil {
-		return err
-	}
-
-	err = f.Sync()
-	if err != nil {
-		return err
-	}
-
-	err = f.Close()
-	if err != nil {
-		return err
+	if err := ioutil.WriteFile(nodefile, finaldata, 0666); err != nil {
+		return fmt.Errorf("failed to write file: %w", err)
 	}
 
+	// TODO(josebalius): should we accept a logger or otherwise not log directly
 	log.Info("Saved p2p node information (%s),  Identity - %v", nodefile, n.publicKey.String())
 
 	return nil
 }
 
-// LoadIdentity loads a specific nodeid from the disk at the given path
-func LoadIdentity(path, nodeid string) (LocalNode, error) {
-	nfd, err := readNodeData(path, nodeid)
+// LoadIdentity loads a specific nodeID from the disk at the given path.
+func LoadIdentity(path, nodeID string) (ln LocalNode, err error) {
+	nfd, err := readNodeData(path, nodeID)
 	if err != nil {
-		return emptyNode, err
+		return ln, fmt.Errorf("failed to read node data: %w", err)
 	}
 
 	return newLocalNodeFromFile(nfd)
 }
 
 // Read node persisted data based on node id.
 func readNodeData(path string, nodeID string) (*nodeFileData, error) {
-
 	nodefile := filepath.Join(path, config.P2PDirectoryPath, config.NodesDirectoryName, nodeID, config.NodeDataFileName)
-
 	if !filesystem.PathExists(nodefile) {
-		return nil, fmt.Errorf("tried to read node from non-existing path (%v)", nodefile)
-	}
-
-	data := bytes.NewBuffer(nil)
-
-	f, err := os.Open(nodefile)
-	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("tried to read node from non-existing path %q", nodefile)
 	}
 
-	_, err = io.Copy(data, f)
-
-	if err != nil {
-		return nil, err
-	}
-
-	err = f.Close()
-
+	b, err := ioutil.ReadFile(nodefile)
```

Comment:
The previous implementation had a potential resource leak. If the `io.Copy` failed, we would fail to close the file handle. `io.ReadFile` can take care of the whole operation for us and we fix two birds with one stone sort of speak.